src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -146,6 +146,243 @@ class Point(int x, int y);
+                // (2,23): error CS1001: Identifier expected
+                // public record A(int i,) { }
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, ")").WithLocation(2, 23)
+                );
+
+            var expectedMembers = new[]
+            {
+                "System.Type A.EqualityContract { get; }",
+                "System.Int32 A.i { get; init; }",
+                "? A. { get; init; }"
+            };
+            AssertEx.Equal(expectedMembers,
+                comp.GetMember<NamedTypeSymbol>("A").GetMembers().OfType<PropertySymbol>().ToTestDisplayStrings());
+
+            AssertEx.Equal(new[] { "A..ctor(System.Int32 i, ? )", "A..ctor(A )" },
+                comp.GetMember<NamedTypeSymbol>("A").Constructors.ToTestDisplayStrings());
+
+            var primaryCtor = comp.GetMember<NamedTypeSymbol>("A").Constructors.First();
+            Assert.Equal("A..ctor(System.Int32 i, ? )", primaryCtor.ToTestDisplayString());
+            Assert.IsType<ParameterSyntax>(primaryCtor.Parameters[1].DeclaringSyntaxReferences.Single().GetSyntax());
+        }
+
+        [Fact, WorkItem(46123, "https://github.com/dotnet/roslyn/issues/46123")]
+        public void IncompletePositionalRecord_WithTrivia()
+        {
+            string source = @"
+public record A(int i, // A
+    // B
+    , /* C */ ) { }
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (2,23): error CS1031: Type expected
+                // public record A(int i, // A
+                Diagnostic(ErrorCode.ERR_TypeExpected, "").WithLocation(2, 23),
+                // (2,23): error CS1001: Identifier expected
+                // public record A(int i, // A
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, "").WithLocation(2, 23),
+                // (4,15): error CS1031: Type expected
+                //     , /* C */ ) { }
+                Diagnostic(ErrorCode.ERR_TypeExpected, ")").WithLocation(4, 15),
+                // (4,15): error CS1001: Identifier expected
+                //     , /* C */ ) { }
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, ")").WithLocation(4, 15),
+                // (4,15): error CS0102: The type 'A' already contains a definition for ''
+                //     , /* C */ ) { }
+                Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "").WithArguments("A", "").WithLocation(4, 15)
+                );
+
+            var primaryCtor = comp.GetMember<NamedTypeSymbol>("A").Constructors.First();
+            Assert.Equal("A..ctor(System.Int32 i, ? , ? )", primaryCtor.ToTestDisplayString());
+            Assert.IsType<ParameterSyntax>(primaryCtor.Parameters[0].DeclaringSyntaxReferences.Single().GetSyntax());
+            Assert.IsType<ParameterSyntax>(primaryCtor.Parameters[1].DeclaringSyntaxReferences.Single().GetSyntax());
+            Assert.IsType<ParameterSyntax>(primaryCtor.Parameters[2].DeclaringSyntaxReferences.Single().GetSyntax());
```

Comment:
Same comment for these.

Reply:
And for the rest of the tests.

---
In reply to: [466051446](https://github.com/dotnet/roslyn/pull/46470#discussion_r466051446) [](ancestors = 466051446)